src/indexset.rs

```diff
@@ -549,6 +550,7 @@ where
     }
 }
 
+#[derive(Clone)]
```

Comment:
This should be a manual `Clone` implementation. No need for `Clone` bounds on any of the type parameters.